### PR TITLE
Setup AMBuild 2.2 + Github Action + Removed threaded nav

### LINKED ISSUE #10: Setup workflows for win & linux builds
Self explanatory. And will provide a rough idea to viewers on how to compile the extension themselves.

Workflow validation will also be required for PRs. This will be done this weekend.

### PR DESCRIPTION
**Pull request by Kenzzer**
We're moving to AMBuild 2.2 so MSCV is properly detected in github actions environment. As well as getting the field ready for CSGO & multi platform support.

`master` branch is now fully protected, every pull requests must be squashed and merged and have at least one review. Also the action from this commit will be required to succeed in order to allow merging of any PRs.

Finally, the threaded nav code has been fully removed, as it prevents compilation of the ext against sm 1.10 and isn't thread safe anyways. It may eventually be revisited in the future.

Close #10

### CODE REVIEW

**Review by KitRifty — CHANGES_REQUESTED** [review]
Mostly everything looks in order to me. The only major thing I can point out is the removal of the the `CNavMesh::AddNavArea()` detour.

Also in addition to the automatic builder, are there plans to implement an automatic release action as well? It would be convenient to have a system in place that would zip up the build artifacts and game data all in one package allowing for easy installation.

File: extension/extension.cpp [hunk 1]
```diff
@@ -40,18 +41,10 @@ SMEXT_LINK(&g_CBaseNPCExt);
 
 IForward *g_pForwardEventKilled = nullptr;
 
-CDetour* g_pNavMeshAddArea = nullptr;
-
 bool (CTraceFilterSimpleHack:: *CTraceFilterSimpleHack::func_ShouldHitEntity)(IHandleEntity *pHandleEntity, int contentsMask) = nullptr;
 
 CUtlMap<int32_t, int32_t> g_EntitiesHooks;
 
-DETOUR_DECL_MEMBER1(CNavMesh_AddNavArea, void, CNavArea*, area)
```
[the remark this answers is not in the document]
Comment (Kenzzer) [reply extension/extension.cpp:49]: Wouldn't it make more sense to re-introduce the detour within navmesh.cpp ?

[the remark this answers is not in the document]
Comment (KitRifty) [reply extension/extension.cpp:49]: Actually yes, it'll help with organization for sure.

**Comment by Kenzzer** [comment]
I'm unfamiliar with release action but yes, this would be very much needed on this repo.

**Comment by Kenzzer** [comment]
Any commit on master will now create a CBaseNPC release. After this lands, it will be very important we update `product.version` file when fixing/adding in new features.

**Review by KitRifty — APPROVED** [review]
Everything in the workflow file looks to be in order, so I'm trusting that you've done some testing with this beforehand. I would like some sort of automation with the versioning so we wouldn't need to edit `product.file` every time, but this can be done in the future in another PR.

Overall, very nice work!

**Pull request merged**